### PR TITLE
Adding ESP32 core selection for MQTT task

### PR DESCRIPTION
**Pull request by vtunr**
Since we can run FreeRTOS on both core, it is interesting to be able to pin MQTT task to one core.

In order to do that, I added one parameter and two choices in KConfig : 

- MQTT_TASK_CORE_SELECTION_ENABLED
- MQTT_USE_CORE_0
- MQTT_USE_CORE_1

 Then it use xTaskCreatePinnedToCore if the option is used.

### CODE REVIEW

File: Kconfig [hunk 1]
```diff
@@ -78,4 +78,19 @@ config MQTT_TASK_STACK_SIZE
     help
         MQTT task stack size
 
+config MQTT_TASK_CORE_SELECTION_ENABLED
+	bool "Enable MQTT task core selection"
+	default f
```
Comment (tuanpmt) [inline Kconfig]: default fast

File: include/mqtt_config.h [hunk 2]
```diff
@@ -59,6 +59,23 @@
 #define MQTT_WSS_DEFAULT_PORT       443
 #endif
 
+#ifdef CONFIG_MQTT_TASK_CORE_SELECTION_ENABLED
+#define MQTT_CORE_SELECTION_ENABLED CONFIG_MQTT_TASK_CORE_SELECTION_ENABLED
+#else
+#define MQTT_CORE_SELECTION_ENABLED 0
```
Comment (tuanpmt) [inline include/mqtt_config.h]: You can use `#define MQTT_CORE_SELECTION_ENABLED CONFIG_MQTT_TASK_CORE_SELECTION_ENABLED` only

File: mqtt_client.c [hunk 3]
```diff
@@ -737,9 +737,18 @@ esp_err_t esp_mqtt_client_start(esp_mqtt_client_handle_t client)
         ESP_LOGE(TAG, "Client has started");
         return ESP_FAIL;
     }
-    if (xTaskCreate(esp_mqtt_task, "mqtt_task", client->config->task_stack, client, client->config->task_prio, NULL) != pdTRUE) {
-        ESP_LOGE(TAG, "Error create mqtt task");
-        return ESP_FAIL;
+    if(MQTT_CORE_SELECTION_ENABLED){
```
Comment (tuanpmt) [inline mqtt_client.c]: And using
```
#if MQTT_CORE_SELECTION_ENABLED
//code for core selection
#else
//code for automatic select
#endif
```

**Comment by vtunr** [comment]
@tuanpmt I updated according to your remarks, I hope everything is good now :)

**Comment by vtunr** [comment]
@tuanpmt I updated according to your remarks, I hope everything is ok now :)

**Pull request merged**